python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -46,6 +46,7 @@ class ComponentGenerateRequest:
 class Component(ABC):
     name: ClassVar[Optional[str]] = None
     params_schema: ClassVar = None
+    rendering_scope: ClassVar = None
```

Comment:
I think we should structure this as a function or, more broadly, think about a world where this is created on demand. We may want to defer imports